src/Akeneo/Bundle/ElasticsearchBundle/Cursor/SearchAfterBoundedCursor.php

```diff
@@ -53,13 +63,55 @@ public function __construct(
             array_push($this->searchAfter, $indexType . '#' . $searchAfterUniqueKey);
         }
 
-        parent::__construct($esClient, $repository, $esQuery, $indexType, $pageSize);
+        $this->items = $this->getNextItems($esQuery);
     }
 
     /**
      * {@inheritdoc}
+     *
+     * @see https://www.elastic.co/guide/en/elasticsearch/reference/5.x/search-request-search-after.html
+     */
+    protected function getNextIdentifiers(array $esQuery)
+    {
+        $esQuery['size'] = $this->getItemsCountToFetch();
+
+        if (0 === $esQuery['size']) {
+            return [];
+        }
+
+        $sort = ['_uid' => 'asc'];
+
+        if (isset($esQuery['sort'])) {
+            $sort = array_merge($esQuery['sort'], $sort);
+        }
+
+        $esQuery['sort'] = $sort;
+
+        if (!empty($this->searchAfter)) {
+            $esQuery['search_after'] = $this->searchAfter;
+        }
+
+        $response = $this->esClient->search($this->indexType, $esQuery);
+        $this->count = $response['hits']['total'];
+
+        $identifiers = [];
+        foreach ($response['hits']['hits'] as $hit) {
+            $identifiers[] = $hit['_source']['identifier'];
+        }
+
+        $lastResult = end($response['hits']['hits']);
+
+        if (false !== $lastResult) {
+            $this->searchAfter = $lastResult['sort'];
+        }
+
+        return $identifiers;
+    }
+
+    /**
+     * @return int
      */
-    protected function getItemsCountToFetch()
+    private function getItemsCountToFetch()
```

Comment:
Line 118: There is a mistake no ? If limit is greater than $this->fetchedItemsCount, $itemsCountToFetch will be negative no ? This is $itemsCountToFetch = $this->limit - $this->fetchedItemsCount instead I think